code/modules/clothing/head/misc.dm

```diff
@@ -288,20 +288,29 @@
 
 	dog_fashion = /datum/dog_fashion/head/sombrero
 
+	greyscale_config = /datum/greyscale_config/sombrero
+
+/obj/item/clothing/head/sombrero/Initialize()
+	. = ..()
+	if(greyscale_colors)
```

Comment:
no need for this, see: https://github.com/tgstation/tgstation/pull/58808/files#diff-c8ab5fbc20de60e202b839834b039649cbb69a1c4b99b27a5e467f3889442ccdR38

tl;dr - just assign greyscale_config_worn, greyscale_config_inhand_left and greyscale_config_inhand_right variables.